src/types/tcb_info.rs

```diff
@@ -0,0 +1,447 @@
+
+            _ => platform_status,
+        }
+    }
+    /// Converge platform TCB status with QE TCB status
+    ///
+    /// This function implements the rules for combining platform and Quote Enclave TCB
+    /// status values, prioritizing the more severe status according to Intel's rules.
+    pub fn converge_tcb_status_with_qe_tcb(
+        platform_status: TcbStatus,
+        qe_status: TcbStatus,
+    ) -> TcbStatus {
+        // Only adjust status if QE is OutOfDate
+        if qe_status != TcbStatus::OutOfDate {
+            return platform_status;
+        }
+
+        match platform_status {
+            // These statuses get overridden to OutOfDate
+            TcbStatus::UpToDate | TcbStatus::SWHardeningNeeded => TcbStatus::OutOfDate,
+
+            // These statuses change to reflect both configuration and outdated problems
+            TcbStatus::ConfigurationNeeded | TcbStatus::ConfigurationAndSWHardeningNeeded => {
+                TcbStatus::OutOfDateConfigurationNeeded
+            },
+
+            // All other statuses remain unchanged
+            _ => platform_status,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TcbLevel {
+    pub tcb: Tcb,
+    pub tcb_date: chrono::DateTime<Utc>,
+    pub tcb_status: TcbStatus,
+    #[serde(rename = "advisoryIDs", skip_serializing_if = "Option::is_none")]
+    pub advisory_ids: Option<Vec<String>>,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Deserialize, Serialize)]
+pub enum TcbStatus {
+    UpToDate,
+    OutOfDate,
+    ConfigurationNeeded,
+    SWHardeningNeeded,
+    ConfigurationAndSWHardeningNeeded,
+    OutOfDateConfigurationNeeded,
+    Revoked,
+    Unspecified,
+}
+
+impl std::fmt::Display for TcbStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TcbStatus::UpToDate => write!(f, "UpToDate"),
+            TcbStatus::OutOfDate => write!(f, "OutOfDate"),
+            TcbStatus::ConfigurationNeeded => write!(f, "ConfigurationNeeded"),
+            TcbStatus::SWHardeningNeeded => write!(f, "SWHardeningNeeded"),
+            TcbStatus::ConfigurationAndSWHardeningNeeded => write!(f, "ConfigurationAndSWHardeningNeeded"),
+            TcbStatus::OutOfDateConfigurationNeeded => write!(f, "OutOfDateConfigurationNeeded"),
+            TcbStatus::Revoked => write!(f, "Revoked"),
+            TcbStatus::Unspecified => write!(f, "Unspecified"),
+        }
+    }
+}
+
+/// Contains information identifying a TcbLevel.
+#[derive(Deserialize, Serialize, PartialEq, Eq, Clone, Debug)]
+#[serde(untagged)]
+pub enum Tcb {
+    V2(TcbV2),
+    V3(TcbV3),
+}
+
+impl Tcb {
+    fn version(&self) -> TcbInfoVersion {
+        match self {
+            Tcb::V2(_) => TcbInfoVersion::V2,
+            Tcb::V3(_) => TcbInfoVersion::V3,
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, PartialEq, Eq, Clone, Debug)]
+pub struct TcbV3 {
```

Comment:
TcbV3 is currently missing the `tdxtcbcomponents` field.

When performing a TCB lookup using PCK Certificate for TDX quotes, you must first match the TCB versions from the SGX components, then perform an additional matching with the TDX components.

See: https://github.com/automata-network/dcap-rs/blob/d847b8f75a493640c4881bdf67775250b6baefab/src/utils/cert.rs#L470-L523

Reply:
Solidity implementation:

https://github.com/automata-network/automata-dcap-attestation/blob/6c80b3bdf97c987ab52bda7fac1c4f11c88fd7b1/evm/contracts/bases/tcb/TCBInfoV3Base.sol#L18-L58

Reply:
Hey @preston4896  after going through the implementation again, i do this already.

https://github.com/udsamani/dcap-rs/blob/main/src/lib.rs#L52-L57

Reply:
Also TDX Components are defined here 

https://github.com/udsamani/dcap-rs/blob/main/src/types/tcb_info.rs#L355-L375

Reply:
Sorry my. bad ! Added a commit. Please check!